ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh

```diff
@@ -579,32 +583,45 @@ CLUSTER_ID=$(cat "${CLUSTER_INFO}" | grep '^ID:' | tr -d '[:space:]' | cut -d ':
 echo "Cluster ${CLUSTER_NAME} is being created with cluster-id: ${CLUSTER_ID}"
 echo -n "${CLUSTER_ID}" > "${CLUSTER_ID_FILE}"
 
-# Watch the hypershift install log
-if [[ "$HOSTED_CP" == "true" ]]; then
-  rosa logs install -c ${CLUSTER_ID} --watch
-fi
-
-echo "Waiting for cluster ready..."
+echo "Capturing installation timers"
 start_time=$(date +"%s")
+dyn_start_time=${start_time}
+CLUSTER_PREVIOUS_STATE=""
+record_cluster "timers.install" "status" "started"
 while true; do
-  sleep 60
   CLUSTER_STATE=$(rosa describe cluster -c "${CLUSTER_ID}" -o json | jq -r '.state')
-  echo "Cluster state: ${CLUSTER_STATE}"
+  current_time=$(date +"%s")
   if [[ "${CLUSTER_STATE}" == "ready" ]]; then
-    echo "Cluster is reported as ready"
+    record_cluster "timers.install" "${CLUSTER_PREVIOUS_STATE}" $(( "${current_time}" - "${dyn_start_time}" ))
+    record_cluster "timers.install" "status" "${CLUSTER_STATE}"
+    echo "Cluster Ready after $(( ${current_time} - ${start_time} )) seconds, exiting"
     break
+  elif [[ "${CLUSTER_STATE}" != "${CLUSTER_PREVIOUS_STATE}" && "${CLUSTER_PREVIOUS_STATE}" != "" ]]; then
+    record_cluster "timers.install" "${CLUSTER_PREVIOUS_STATE}" $(( "${current_time}" - "${dyn_start_time}" ))
+    record_cluster "timers.install" "status" "${CLUSTER_STATE}"
+    echo "Cluster transitioned from ${CLUSTER_PREVIOUS_STATE} to ${CLUSTER_STATE} after $(( ${current_time} - ${dyn_start_time} )) seconds"
+    dyn_start_time=$(date +"%s")
+  else
+    if (( "${current_time}" - "${dyn_start_time}" >= "${CLUSTER_TIMEOUT}" )); then
+      echo "error: Cluster not ready after ${CLUSTER_TIMEOUT}seconds. State: ${CLUSTER_STATE}"
+      record_cluster "timers.install" "status" "${CLUSTER_STATE}"
+      exit 1
+    else
+      if [[ ${CLUSTER_STATE} == "installing" ]]; then
+      	echo "Cluster on ${CLUSTER_STATE} state, waiting 60 second for the next check..."
+      	sleep 60
+      else
+        echo "Cluster on ${CLUSTER_STATE} state, waiting 1 second for the next check..."
+        sleep 1
+      fi
+    fi
```

Comment:
Meaningless to check each status's timer. For provision, what we want is the whole installation time.

Reply:
preflight timers are also important, as we have detected regressions on aws checks done before cluster is installed, during the pending and the validating phases

Reply:
You are using the same item `timers.install` to record the timer of all of the status. At last, if everything goes well, the timers.install is for the 'Ready' status.  You do not record the preflight check timer.

And in this block, the changes are written complex. 
1. First, in a loop, we'd better always put the exit at the top, in an individual level-1 `if`.
2. Always use `if` instead of `if .. else`

Reply:
I am not using the same field to record the status, timers.install is a dictionary and each status has its own field

the final status of the timers dictionary will be something like this:

```
  "timers": {
    "install": {
      "status": "ready",
      "waiting": "30",
      "validating": "5",
      "pending": "69",
      "installing": "1935",
      "total_install": "2039"
    },
    "co_wait_time": "0",
    "nodes_ready": "608",
    "destroy": "424",
    "sts_destroy": "18"
  }
```

Inside timers.install, you get a status field with the last status reached by the cluster installation, and a field with the time dedicated to each transition since the cluster is created.

A rosa cluster transition on this chain waiting > validating > pending > installing > ready

Reply:
Keep the extreme condition(timeout) at the main if block and error state.

Reply:
done